sdk/core/perf/inc/azure/perf/random_stream.hpp

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief A random stream of any specific size. Useful for test cases.
+ *
+ */
+
+#pragma once
+
+#include <azure/core/io/body_stream.hpp>
+
+#include <memory>
+#include <random>
+
+namespace Azure { namespace Perf {
+
+  /**
+   * @brief A random stream of any specific size. Useful for test cases.
+   *
+   */
+  class RandomStream {
+  private:
+    /**
+     * @brief Wraps a stream and keep reading bytes from it by rewinding it until some length.
+     *
+     * @note Enables to create a stream with huge size by re-using a small buffer.
+     *
+     */
+    class CircularStream : public Azure::Core::IO::BodyStream {
```

Comment:
Why nest it?

Reply:
The CircularStream can be considered as an implementation detail of the RandomStream.

Is currently the RandomStream strategy to provide a size-n stream.  There's no need, for now, to expose the CircularStream as public from the perf-fw.

Actually, I might be able to move the entire class to a cpp file to keep it totally as an implementation detail.

The only benefit of having it here in the header is that we can have the `Create` function implemented here :)